crates/forge_analyzer/src/checkers.rs

```diff
@@ -446,3 +459,597 @@ impl IntoVuln for AuthNVuln {
+                                let mut all_values = prev_values.clone();
+                                all_values.push(const_value);
+                                let value = Value::Phi(all_values);
+                                self.variables_from_defid.insert(*defid, value.clone());
+                            } else {
+                                let value = Value::Const(Const::Object(class.clone()));
+                                self.variables_from_defid.insert(*defid, value.clone());
+                            }
+                        } else if let Some(value) = self.variables_from_defid.get(defid) {
+                            /* add value from defid */
+                        }
+                    }
+                }
+                _ => {}
+            },
+        }
+    }
+
+    fn join_term<C: crate::interp::Checker<'cx, State = Self::State>>(
+        &mut self,
+        interp: &mut Interp<'cx, C>,
+        def: DefId,
+        block: &'cx BasicBlock,
+        state: Self::State,
+        worklist: &mut WorkList<DefId, BasicBlockId>,
+    ) {
+        self.super_join_term(interp, def, block, state, worklist);
+        for (def, arguments) in self.needs_call.drain(..) {
+            worklist.push_front_blocks(interp.env(), def, arguments.clone());
+            interp.callstack_arguments.push(arguments.clone());
+        }
+    }
+}
+
+fn resolve_var_from_operand(operand: &Operand) -> Option<VarId> {
+    if let Operand::Var(var) = operand {
+        if let Base::Var(varid) = var.base {
+            return Some(varid);
+        }
+    }
+    None
+}
+
+fn resolve_literal_from_operand(operand: &Operand) -> Option<Literal> {
+    if let Operand::Lit(lit) = operand {
+        return Some(lit.clone());
+    }
+    None
+}
+
+fn get_varid_from_defid(varkind: &VarKind) -> Option<DefId> {
+    match varkind {
+        VarKind::GlobalRef(defid) => Some(defid.clone()),
+        VarKind::LocalDef(defid) => Some(defid.clone()),
+        VarKind::Arg(defid) => Some(defid.clone()),
+        VarKind::AnonClosure(defid) => Some(defid.clone()),
+        VarKind::Temp { parent } => parent.clone(),
+        _ => None,
+    }
+}
+
+fn convert_operand_to_raw(operand: &Operand) -> Option<String> {
+    if let Operand::Lit(lit) = operand {
+        convert_lit_to_raw(lit)
+    } else {
+        None
+    }
+}
+
+fn convert_lit_to_raw(lit: &Literal) -> Option<String> {
+    match lit {
+        Literal::BigInt(bigint) => Some(bigint.to_string()),
+        Literal::Number(num) => Some(num.to_string()),
+        Literal::Str(str) => Some(str.to_string()),
+        _ => None,
+    }
+}
+
+fn find_member_of_obj(member: &str, obj: &Class) -> Option<DefId> {
+    for (mem, memdefid) in &obj.pub_members {
+        if mem == member {
+            return Some(memdefid.clone());
+        }
+    }
+    None
+}
+
+fn add_elements_to_intrinsic_struct(value: &Value, args: &mut Option<Vec<String>>) {
```

Comment:
It might be better to have `args` just be a `&mut Vec`, since this function calls `unwrap()` on it anyways.